test/built-ins/Array/prototype/includes/values-are-not-cached.js

```diff
@@ -0,0 +1,52 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-array.prototype.includes
+description: indexed values are not cached
+info: |
+  22.1.3.11 Array.prototype.includes ( searchElement [ , fromIndex ] )
+
+  ...
+  7. Repeat, while k < len
+    a. Let elementK be the result of ? Get(O, ! ToString(k)).
+  ...
+includes: [compareArray.js]
+---*/
+
+function getCleanObj() {
+  var obj = {};
+  Object.defineProperty(obj, "length", {
+    get: function() {
+      Object.defineProperty(obj, "0", {
+        get: function() {
+          Object.defineProperty(obj, "1", {
+            get: function() {
+              return "ecma262";
+            }
+          });
+          Object.defineProperty(obj, "2", {
+            get: function() {
+              return "cake";
+            }
+          });
+          return "tc39";
+        }
+      });
+      return 2;
+    }
+  });
+
+  return obj;
+}
+
+var obj;
+obj = getCleanObj();
+assert.sameValue([].includes.call(obj, "tc39"), true, "'tc39' is true");
```

Comment:
The iteration halt on match is covered here: https://github.com/tc39/test262/pull/643/files#diff-e26af1e445b843aadb5c20d15bf12f39R41